azurerm/resource_arm_nat_gateway.go

```diff
@@ -0,0 +1,306 @@
+				return fmt.Errorf("Error checking for present of existing NAT Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+			}
+		}
+		if !utils.ResponseWasNotFound(resp.Response) {
+			return tf.ImportAsExistsError("azurerm_nat_gateway", *resp.ID)
+		}
+	}
+
+	location := azure.NormalizeLocation(d.Get("location").(string))
+	idleTimeoutInMinutes := d.Get("idle_timeout_in_minutes").(int)
+	publicIpAddressIds := d.Get("public_ip_address_ids").([]interface{})
+	publicIpPrefixIds := d.Get("public_ip_prefix_ids").([]interface{})
+	skuName := d.Get("sku_name").(string)
+	zones := d.Get("zones").([]interface{})
+	t := d.Get("tags").(map[string]interface{})
+
+	parameters := network.NatGateway{
+		Location: utils.String(location),
+		NatGatewayPropertiesFormat: &network.NatGatewayPropertiesFormat{
+			IdleTimeoutInMinutes: utils.Int32(int32(idleTimeoutInMinutes)),
+			PublicIPAddresses:    expandArmNatGatewaySubResourceID(publicIpAddressIds),
+			PublicIPPrefixes:     expandArmNatGatewaySubResourceID(publicIpPrefixIds),
+		},
+		Sku: &network.NatGatewaySku{
+			Name: network.NatGatewaySkuName(skuName),
+		},
+		Tags:  tags.Expand(t),
+		Zones: utils.ExpandStringSlice(zones),
+	}
+
+	future, err := client.CreateOrUpdate(ctx, resourceGroup, name, parameters)
+	if err != nil {
+		return fmt.Errorf("Error creating NAT Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("Error waiting for creation of NAT Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		return fmt.Errorf("Error retrieving NAT Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read NAT Gateway %q (Resource Group %q) ID", name, resourceGroup)
+	}
+	d.SetId(*resp.ID)
+
+	return resourceArmNatGatewayRead(d, meta)
+}
+
+func resourceArmNatGatewayRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).Network.NatGatewaysClient
+	ctx := meta.(*ArmClient).StopContext
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+	resourceGroup := id.ResourceGroup
+	name := id.Path["natGateways"]
+
+	resp, err := client.Get(ctx, resourceGroup, name, "")
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] NAT Gateway %q does not exist - removing from state", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading NAT Gateway %q (Resource Group %q): %+v", name, resourceGroup, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("sku_name", resp.Sku.Name)
+	d.Set("resource_group_name", resourceGroup)
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+	if natGatewayPropertiesFormat := resp.NatGatewayPropertiesFormat; natGatewayPropertiesFormat != nil {
```

Comment:
can we update this to be `props`?
```suggestion
	if props:= resp.NatGatewayPropertiesFormat; props != nil {
```

Reply:
Done